kymatio/scattering1d/backend/torch_skcuda_backend.py

```diff
@@ -62,29 +62,29 @@ def __call__(self, x):
         if not x.is_cuda and self.backend=='skcuda':
             raise TypeError('Use the torch backend (without skcuda) for CPU tensors.')
 
-        out = x.new(x.shape)
+        out = x.new(x.shape[:-1] + (1,))
 
         if not x.is_contiguous():
             raise RuntimeError('Input should be contiguous.')
 
         if not _is_complex(x):
-            raise TypeError('The input and outputs should be complex.')
+            raise TypeError('The input should be complex.')
 
         kernel = """
         extern "C"
-        __global__ void abs_complex_value(const ${dtype} * x, ${dtype}2 * z, int n)
+        __global__ void abs_complex_value(const ${dtype} * x, ${dtype} * z, int n)
         {
             int i = blockIdx.x * blockDim.x + threadIdx.x;
         if (i >= n)
             return;
-        z[i] = make_${dtype}2(normf(2, x + 2*i), 0);
+        z[i] = normf(2, x + 2*i);
 
         }
         """
         fabs = load_kernel('abs_complex_value', kernel, dtype=get_dtype(x))
-        fabs(grid=(self.get_blocks(int(out.nelement())//2), 1, 1),
+        fabs(grid=(self.get_blocks(int(out.nelement())), 1, 1),
              block=(self.CUDA_NUM_THREADS, 1, 1),
-             args=[x.data_ptr(), out.data_ptr(), out.numel() // 2],
+             args=[x.data_ptr(), out.data_ptr(), out.numel()],
```

Comment:
Can you explain this change and the one above?

Reply:
I believe we want to specify the number of elements we are working with.

Reply:
Should i do x.numel() // 2 ?

Reply:
I don't fully grasp this code. @edouardoyallon halp